packages/trader/src/App/Components/Elements/ContractAudit/contract-audit.tsx

```diff
@@ -1,35 +1,48 @@
-import PropTypes from 'prop-types';
 import React from 'react';
 import { Tabs } from '@deriv/components';
 import { localize } from '@deriv/translations';
 import { WS } from '@deriv/shared';
-import ContractDetails from './contract-details';
+import { useTraderStore } from 'Stores/useTraderStores';
+import ContractDetails from './contract-details.jsx';
 import ContractHistory from './contract-history.jsx';
 
+type TContractUpdateHistory = [] | { order_date: number }[];
+
+type TContractAudit = Pick<ReturnType<typeof useTraderStore>, 'is_accumulator' | 'is_turbos' | 'is_multiplier'> & {
+    contract_update_history: TContractUpdateHistory;
+    has_result: boolean;
+    toggleHistoryTab: (state_change?: boolean) => void;
+};
+
+type TResponse = {
+    contract_update_history: TContractUpdateHistory;
+};
+
 const ContractAudit = ({
     contract_update_history,
     has_result,
     is_accumulator,
     is_multiplier,
-    is_smarttrader_contract,
     is_turbos,
     toggleHistoryTab,
     ...props
-}) => {
+}: TContractAudit) => {
+    //@ts-expect-error until parent component will be typescript migrated in order to verify props
     const { contract_id, currency } = props.contract_info;
-    const [update_history, setUpdateHistory] = React.useState([]);
+    const [update_history, setUpdateHistory] = React.useState<TContractUpdateHistory>([]);
 
-    const getSortedUpdateHistory = history => history.sort((a, b) => b.order_date - a.order_date);
+    const getSortedUpdateHistory = (history: TContractUpdateHistory) =>
+        history.sort((a, b) => b.order_date - a.order_date);
 
     React.useEffect(() => {
         if (!!contract_update_history.length && contract_update_history.length > update_history.length)
             setUpdateHistory(getSortedUpdateHistory(contract_update_history));
     }, [contract_update_history, update_history]);
 
-    const onTabItemClick = tab_index => {
-        toggleHistoryTab(tab_index);
+    const onTabItemClick = (tab_index: number) => {
+        toggleHistoryTab(!!tab_index);
```

Comment:
In ui-store `toggleHistoryTab` accepts boolean so I converted `tab_index` to it.